executor/adapter.go

```diff
@@ -437,10 +437,12 @@ func (a *ExecStmt) Exec(ctx context.Context) (_ sqlexec.RecordSet, err error) {
 	sctx := a.Ctx
 	ctx = util.SetSessionID(ctx, sctx.GetSessionVars().ConnectionID)
 	if _, ok := a.Plan.(*plannercore.Analyze); ok && sctx.GetSessionVars().InRestrictedSQL {
-		oriStats, _ := sctx.GetSessionVars().GetSystemVar(variable.TiDBBuildStatsConcurrency)
+		oriStats, err := sctx.GetSessionVars().GetSessionOrGlobalSystemVar(variable.TiDBBuildStatsConcurrency)
+		terror.Log(err)
```

Comment:
> GetSystemVar returns empty values for both system variables in an internal session

I think we should try to avoid it, if it is not trivial to do so, we could use the default value as 'oriStats' here.

Reply:
It seems not trivial to use the global system variable value to initialize the session system variable for internal sessions because we need the first internal session to read the global system variable value from disk. The two quick-fix ways are:
1. When `GetSystemVar` returns empty value, use the default value.
2. Use `GetSessionOrGlobalSystemVar`(use the session system variable value if it exists, otherwise use the global system variable value).
Which ways do you prefer?

Reply:
I think to be default value is fine

Reply:
defaults are safer for internal sessions. Probably outside of the scope here, but we should even consider making internal sessions only use defaults unless a variable has a special flag saying it should apply to internal sessions.

Consider the case that the user changes the Txn Isolation, Optimistic/Pessimistic locking, change in lock wait timeout etc. These could affect how statements execute quite a lot, resulting in bugs.

Reply:
Changed to use default value. PTAL @xiongjiwei @morgo @bb7133